### DISCUSSION TITLE
How to train Spacy NER model on multiple GPUs ?

### DISCUSSION THREAD

**Discussion by rs-pawanmethre** [explosion/spaCy#12764] [Help: Other Questions] [answered]
I am training my model using this command : python3 -m spacy train config.cfg  --output output --paths.train train_data.spacy --paths.dev test_data.spacy --gpu-id 0

but my question is how to train model on multiple gpu as want to achieve data parallelism, i want to give more data for training.
Please kindly help me

**Comment by danieldk** [comment, marked as answer]
Thanks for your question. We do not support multiple-GPU training yet.

**Reply by rs-pawanmethre** [reply]
Thanks @danieldk  for your prompt reply , i had one follow up question. How to train model on more data then for spacy with 16GB GPU ?

**Reply by danieldk** [reply]
The first question is what your bottleneck is. The amount of GPU memory is typically not an issue when training with more data, except if you would increase the batch size (which could lead to different training dynamics as well). If the bottleneck is training speed, the next step depends on what kind of model you are training.

**Transformer**: if your GPU supports FP16, you can get better training performance by enabling mixed precision training using the `mixed_precision` option.

**Convolution network (CNN)**: with a reasonably modern GPU, the GPU is usually not the bottleneck for convolution models. GPU performance is bottlenecked by the large number of CPU <-> GPU transfers, particularly in the parser and NER and you can sometimes get better performance by training on the CPU with very fast matrix multiplication (eg. when using the AMX units on modern Apple Silicon Macs through `thinc-apple-ops`.

**Reply by rs-pawanmethre** [reply]
Thank You.
